src/scripts/templates/gift_card.js

```diff
@@ -1,32 +1,35 @@
+import QRCode from 'qrcode';
+
 /**
  * Gift Card Template Script
  * ------------------------------------------------------------------------------
  * A file that contains scripts highly couple code to the Gift Card template.
  */
 
-import $ from 'jquery';
-
-const config = {
-  qrCode: '#QrCode',
-  printButton: '#PrintGiftCard',
-  giftCardCode: '.giftcard__code',
+const selectors = {
+  qrCodeCanvas: '[data-gift-card-qr]',
+  printButton: '[data-gift-card-print]',
+  giftCardCode: '[data-gift-card-digits]',
 };
 
-// new QRCode($qrCode[0], {
-//   text: $qrCode.attr('data-identifier'),
-//   width: 120,
-//   height: 120,
-// });
+// This is the QR code that allows customers to use at a POS
+document.querySelectorAll(selectors.qrCodeCanvas).forEach((qrCodeCanvas) => {
+  QRCode.toCanvas(qrCodeCanvas, qrCodeCanvas.getAttribute('data-identifier'));
+});
 
-$(config.printButton).on('click', () => {
-  window.print();
+document.querySelectorAll(selectors.printButton).forEach((printButton) => {
+  printButton.addEventListener('click', () => {
```

Comment:
Can be written more concisely as:
```
printButton.addEventListener('click', window.print);
```

Reply:
Not sure why it didn't work as window.print 

Demo URL: https://harshaltest1.myshopify.com/?fts=0&preview_theme_id=61223436388

Once navigate, modify endpoint to : /gift_cards/17564827748/e1081565e96f2cb089eb1200c9b54d62

It works with arrow function though

Demo URL: https://harshaltest1.myshopify.com/?fts=0&preview_theme_id=61234446436

Once navigate, modify endpoint to : /gift_cards/17564827748/e1081565e96f2cb089eb1200c9b54d62

Reply:
Weird https://stackoverflow.com/questions/26217540/window-print-as-direct-jquery-callback-illegal-invocation. Oh well thanks for trying!